### PR TITLE
Real robot changes

### PR DESCRIPTION
**Pull request by bmagyar**

### CODE REVIEW

File: piper_bringup/launch/piper.launch.xml [hunk 1]
```diff
@@ -22,4 +22,13 @@
           args="joint_trajectory_controller gripper_controller joint_state_broadcaster -c /controller_manager" />
 
     <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share piper_bringup)/rviz/robotmodel.rviz" output="both"/>
+
+    <!-- official python CAN driver -->
+    <node pkg="piper" exec="piper_single_ctrl" output="both">
```
Comment (MarqRazz) [inline piper_bringup/launch/piper.launch.xml]: Only need to run this when `mock_hardware=false`

File: piper_description/urdf/piper.ros2_control.xacro [hunk 2]
```diff
@@ -8,6 +8,7 @@
     mock_hardware:=false
     fake_sensor_commands:=false
     sim_gazebo:=false
+    sim_ignition:=false
```
Comment (MarqRazz) [inline piper_description/urdf/piper.ros2_control.xacro]: `ignition` is a deprecated term. I use `sim_gazebo` here.

File: piper_bringup/launch/piper.launch.xml [hunk 3]
```diff
@@ -22,4 +22,13 @@
...
+        <param name="can_port" value="can0"/>
+        <param name="auto_enable" value="true"/>
+        <param name="rviz_ctrl_flag" value="false"/>
+        <param name="gripper_exist" value="true"/>
```
Comment (MarqRazz) [inline piper_bringup/launch/piper.launch.xml]: Can we add a parameter here for the command and state topics?

File: piper_bringup/package.xml [hunk 4]
```diff
@@ -9,12 +9,13 @@
...
 
-  <exec_depend>rviz2</exec_depend>
   <exec_depend>controller_manager</exec_depend>
   <exec_depend>gz_ros2_control</exec_depend>
-  <exec_depend>parallel_gripper_controller</exec_depend>
   <exec_depend>joint_state_broadcaster</exec_depend>
   <exec_depend>joint_trajectory_controller</exec_depend>
+  <exec_depend>parallel_gripper_controller</exec_depend>
+  <exec_depend>rviz2</exec_depend>
+  <exec_depend>topic_based_ros2_control</exec_depend>
```
Comment (MarqRazz) [inline piper_bringup/package.xml:19]: This also now depends on `piper_single_ctrl`. Can we copy the package over to this repo?

File: piper_description/urdf/piper.ros2_control.xacro (hunk 2 above)
Comment (bmagyar) [reply piper_description/urdf/piper.ros2_control.xacro]: it is used in the macro down below, I added it for it to not crash

File: piper_bringup/launch/piper.launch.xml (hunk 3 above)
Comment (bmagyar) [reply piper_bringup/launch/piper.launch.xml]: why? I think it is more bulletproof to handle the two topics via a single parameter

File: piper_bringup/launch/piper.launch.xml (hunk 1 above)
Comment (bmagyar) [reply piper_bringup/launch/piper.launch.xml]: good point! I added that locally already

File: piper_bringup/package.xml (hunk 4 above)
Comment (bmagyar) [reply piper_bringup/package.xml:19]: I'll add a copy

File: piper_description/urdf/piper.ros2_control.xacro (hunk 2 above)
Comment (MarqRazz) [reply piper_description/urdf/piper.ros2_control.xacro]: Yep it needs to be removed from line 32.

File: piper_bringup/launch/piper.launch.xml (hunk 3 above)
Comment (MarqRazz) [reply piper_bringup/launch/piper.launch.xml]: To control multiple arms at the same time.

File: piper_description/urdf/piper_macro.xacro [hunk 5]
```diff
@@ -29,20 +29,18 @@
...
-        prefix="${prefix}"
-        include_gripper="${include_gripper}"
-        mock_hardware="${mock_hardware}"
-        sim_isaac="${sim_isaac}"
-        isaac_joint_commands="${isaac_joint_commands}"
-        isaac_joint_states="${isaac_joint_states}"
-        sim_gazebo="${sim_gazebo}"
-        initial_positions="${initial_positions}"/>
-    </xacro:if>
+    <!-- ros2 control is only used for simulation -->
```
Comment (MarqRazz) [inline piper_description/urdf/piper_macro.xacro]: Can we refactor or remove this comment.

File: piper_bringup/launch/piper.launch.xml (hunk 3 above)
Comment (MarqRazz) [reply piper_bringup/launch/piper.launch.xml]: This and moving the driver over to the repo can be a future addition.

File: piper_description/urdf/piper.ros2_control.xacro (hunk 2 above)
Comment (bmagyar) [reply piper_description/urdf/piper.ros2_control.xacro:11]:
```suggestion

```

File: piper_bringup/launch/piper.launch.xml (hunk 3 above)
Comment (bmagyar) [reply piper_bringup/launch/piper.launch.xml]: aye let's do that in a new PR, good point with the 2 arms

File: piper_description/urdf/piper_macro.xacro (hunk 5 above)
Comment (bmagyar) [reply piper_description/urdf/piper_macro.xacro]:
```suggestion

```

**Review by MarqRazz — APPROVED** [review]
After you add the `mock_hardware=false` guard to the launch file and update the conflict I think this is good to go.

**Comment by bmagyar** [comment]
done!

**Pull request merged**